bezier-rs/lib/src/subpath/mod.rs

```diff
@@ -0,0 +1,56 @@
+mod core;
+mod lookup;
+mod structs;
+pub use structs::*;
+
+use crate::Bezier;
+
+use std::ops::{Index, IndexMut};
+
+/// Structure used to represent a path composed of [Bezier] curves.
+pub struct Subpath {
+	manipulator_groups: Vec<ManipulatorGroup>,
+	closed: bool,
+}
+
+impl Index<usize> for Subpath {
+	type Output = ManipulatorGroup;
+
+	fn index(&self, index: usize) -> &Self::Output {
+		assert!(index < self.len());
+		&self.manipulator_groups[index]
+	}
+}
+
+impl IndexMut<usize> for Subpath {
+	fn index_mut(&mut self, index: usize) -> &mut Self::Output {
+		assert!(index < self.len());
+		&mut self.manipulator_groups[index]
+	}
+}
+
+impl Iterator for SubpathIter<'_> {
+	type Item = Bezier;
+
+	fn next(&mut self) -> Option<Self::Item> {
+		if self.index >= self.sub_path.len() - 1 + (self.sub_path.closed as usize) {
```

Comment:
A few descriptive comments regarding the invariants / considerations of this iterator might be worth adding, would help for future clarity.

Reply:
I added a quick comment on top of the next function, but I'm not 100% what you had in mind for invariants. I feel like its generally assumed that iterator behavior is undefined if the object over which it is manipulating is modified.